File: Hackerearth/DP/studious_jhool.cpp
/*
				C++
				encoding:UTF-8
				Modified: <14/Apr/2019 12:48:20 AM>

				✪ H4WK3yE乡
				Mohd. Farhan Tahir
				Indian Institute Of Information Technology (IIIT),Gwalior
*/

#include <bits/stdc++.h>

using namespace std;

// clang-format off

#define		 ll                     int64_t
#define		 mod                    1000000007
#define		 ve                     vector
#define		 pb                     push_back
#define		 umap                   unordered_map
#define		 pq                     priority_queue
#define		 endl                   "\n";
#define		 ff                     first
#define		 ss                     second
#define		 pii                    pair<int, int>
#define		 len(v)                 int(v.size())
#define		 all(v)                 v.begin(), v.end()
#define		 reset(a, b)            memset(a, b, sizeof(a));
#define		 sum(v)                 accumulate(v.begin(), v.end(), (ll)0);
#define		 pr1(x)                 cout<<x<<"\n"
#define		 pr2(x, y)              cout<<x<<" "<<y<<"\n"
#define		 pr3(x, y, z)           cout<<x<<" "<<y<<" "<<z<<"\n"
#define		 prv(v, s, n)           fr(i, s, n) cout<<v[i]<<" "; cout<<"\n"
#define		 fr(i, s, n)            for (int i = s ; i < n ; ++i)
#define		 dfr(i, s, n)           for (int i = s ; i > n ; --i)

ll powermod(ll _a,ll _b,ll _m) {ll _r=1;while(_b){if(_b%2==1)_r=(_r*_a)%_m;_b/=2;_a=(_a*_a)%_m;}return _r;}

// clang-format on

int dp[200];

int solve (int x) {
	if (x < 0) return 1e9;
	if (x == 0) return 0;
	if (dp[x] != -1) return dp[x];
	return dp[x] = 1 + min (solve (x - 10), solve (x - 12));
}

int main () {
	ios_base::sync_with_stdio (false), cin.tie (0);
	int tc;
	cin >> tc;
	while (tc--) {
		reset (dp, -1);
		int n;
		cin >> n;
		int ans = solve (n);
		if (ans >= 1e9)
			pr1 (-1);
		else
			pr1 (ans);
	}
	return 0;
}

File: Hackerearth/Dynamic Programming/Introduction to Dynamic Programming/studious_jhool.py
#	!/bin/env python3
#	encoding: UTF-8
#	Modified: <31/Dec/2018 12:07:42 PM>


#	✪ H4WK3yE乡
#	Mohd. Farhan Tahir
#	Indian Institute Of Information Technology (IIIT),Gwalior

#	Question Link
#
#

# ///==========Libraries, Constants and Functions=============///


import sys
sys.setrecursionlimit(20000)

inf = float("inf")
mod = 1000000007


def get_array(): return list(map(int, sys.stdin.readline().split()))


def get_ints(): return map(int, sys.stdin.readline().split())


def input(): return sys.stdin.readline()

# ///==========MAIN=============///


def recurse(i, dp):
    if i < 0:
        return inf
    if i == 0:
        return 0
    if dp[i] != inf:
        return dp[i]
    dp[i] = min(1 + recurse(i - 10, dp), 1 + recurse(i - 12, dp))
    return dp[i]


def main():
    for tc in range(int(input())):
        dp = [inf for _ in range(115)]
        n = int(input())
        ans = recurse(n, dp)
        print(ans) if ans != inf else print(-1)


if __name__ == "__main__":
    main()
